From 3971bceb15f7fc4d83e8fbd209945269f657a070 Mon Sep 17 00:00:00 2001
From: Olivia Lee <olivia.lee@collabora.com>
Date: Fri, 14 Nov 2025 20:29:26 -0800
Subject: [PATCH] panvk/csf: fix uninitialized read in draw context

We check fn_set_fbds_provoking_vertex_stride == 0 to determine whether a
previous function variant has already been allocated, so this value must
be initialized to zero before we start the loop. We could fix this by
explicitly initializing just that field, but I figure it's simpler and
safer to just zero-initialize the whole struct.

Signed-off-by: Olivia Lee <olivia.lee@collabora.com>
Fixes: 885805560f9 ("panvk/csf: fix case where vk_meta is used before PROVOKING_VERTEX_MODE_LAST")
Reviewed-by: Eric R. Smith <eric.smith@collabora.com>
Reviewed-by: Boris Brezillon <boris.brezillon@collabora.com>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/38458>
(cherry picked from commit e899bc8be8cb725ca7a4a76ffccf2888db3db751)
---
 .pick_status.json                           | 2 +-
 src/panfrost/vulkan/csf/panvk_vX_cmd_draw.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/.pick_status.json b/.pick_status.json
index 0a8ebd52388d8..9f179c1b8fada 100644
--- a/.pick_status.json
+++ b/.pick_status.json
@@ -1114,7 +1114,7 @@
         "description": "panvk/csf: fix uninitialized read in draw context",
         "nominated": true,
         "nomination_type": 2,
-        "resolution": 0,
+        "resolution": 1,
         "main_sha": null,
         "because_sha": "885805560f97c0b777d47db550ba51234b603a50",
         "notes": null
diff --git a/src/panfrost/vulkan/csf/panvk_vX_cmd_draw.c b/src/panfrost/vulkan/csf/panvk_vX_cmd_draw.c
index fe3d557e848a8..efc8dc5d90d3b 100644
--- a/src/panfrost/vulkan/csf/panvk_vX_cmd_draw.c
+++ b/src/panfrost/vulkan/csf/panvk_vX_cmd_draw.c
@@ -137,7 +137,7 @@ calc_fn_set_fbds_provoking_vertex_idx(struct panvk_cmd_buffer *cmdbuf)
 VkResult
 panvk_per_arch(device_draw_context_init)(struct panvk_device *dev)
 {
-   dev->draw_ctx = vk_alloc(&dev->vk.alloc,
+   dev->draw_ctx = vk_zalloc(&dev->vk.alloc,
             sizeof(struct panvk_device_draw_context),
             _Alignof(struct panvk_device_draw_context),
             VK_SYSTEM_ALLOCATION_SCOPE_DEVICE);
-- 
GitLab

